src/components/drop-down/drop-down.tsx

```diff
@@ -96,19 +94,19 @@ export class DropDown extends React.PureComponent<DropDownProps, DropDownState>
 
     private toggleDropdown = () => {
         if (!this.props.disabled) {
-            this.setState({open: !this.state.open});
+            this.props.onOpenStateChange && this.props.onOpenStateChange({value: !this.props.open});
         }
     }
 
     private openDropdown() {
-        if (!this.props.disabled) {
-            this.setState({open: true});
+        if (!this.props.disabled && !this.props.open) {
```

Comment:
Isn't it simpler just to trigger `onOpenStateChange({ value: true })` ? :-)

Reply:
it is! changing :)